app/models/branch.js

```diff
@@ -20,3 +20,18 @@ export default Model.extend({
     }
   })
 });
+
+Branch.reopenClass({
+  search(store, name, repositoryId) {
+    return store.query('branch', {
+      repository_id: repositoryId,
+      data: {
+        name: name,
+        sort_by: 'repository.name',
+        limit: 10
+      }
```

Comment:
1. There's no need to nest properties under `data` - serializer will handle that.  
2. `sort_by` appear to not be working properly (especially given `repository.name`, why sort branches by repository name?) - it just doesn't affect the results, they are not sorted anyways, so it can be removed. 
3. Limit does not make sense here too, API has default limit and we should fetch as much as it can give us (why limit users in dropdown?). 
4. Although, what we do need here is a filter by `exists_on_github`, because otherwise create cron request will return an error.

As a result, payload should be like this:
```js
    return store.query('branch', {
      repository_id: repositoryId,
      name,
      exists_on_github: true
    });
```
As you may notice, it's already not a general "branch search", it's "branch search for cron jobs". That's why I suggested to move the request to the respective component - no need to expose specific request as a global method.

Reply:
If sort is not working correctly, I'll fis it on api side. Limit make sense because we want only 10 branches per result

Reply:
1. Yes, I'm sure, serializer does that job.
3. I didn't know about the limit. It just doesn't feel right from the UX perspective. Should we rise the conversation about it in the Slack channel?

Reply:
1. doesnt work witthout the wrap
2. I think that we should decide about it, 10 in my opinion is ok

Reply:
1. Please refer to this comment: https://github.com/travis-ci/travis-web/pull/2024#discussion_r288986604
2. I agree, 10 sounds good to me. Anyways, we can easily increase it any time later in case of customers' feedback.